engine/apps/user_management/utils.py

```diff
@@ -0,0 +1,35 @@
+import typing
+
+from apps.user_management.constants import (
+    AlertGroupTableColumn,
+    AlertGroupTableColumns,
+    AlertGroupTableColumnTypeChoices,
+    AlertGroupTableDefaultColumnChoices,
+)
+
+if typing.TYPE_CHECKING:
+    from apps.user_management.models import User
+
+
+def default_columns() -> typing.List[AlertGroupTableColumn]:
+    columns = [
+        {"name": column.label, "id": column.value, "type": AlertGroupTableColumnTypeChoices.DEFAULT.value}
+        for column in AlertGroupTableDefaultColumnChoices
+    ]
+    return columns
+
+
+def alert_group_table_user_settings(user: "User") -> AlertGroupTableColumns:
+    organization_columns = user.organization.alert_group_table_columns
+    visible_columns: typing.List[AlertGroupTableColumn]
```

Comment:
Maybe type [`user.alert_groups_table_selected_columns`](https://github.com/grafana/oncall/pull/3281/files#diff-c569330343b5bb0d5520d3c300e93b6d3d0fd07c73509ab3354161eda10e2be4R240) to be `list[AlertGroupTableColumns]` so this additional annotation wouldn't be needed?

Reply:
Sorry, didn't get it, where do you mean to type it?

Reply:
```python3
alert_groups_table_selected_columns: list[AlertGroupTableColumn] | None = models.JSONField(default=None, null=True)
```